File: example/t.py
#!/usr/bin/python
#coding=utf-8

import sys
reload(sys)
sys.setdefaultencoding('utf-8')
sys.path.append(r'../libs')

import re
import json
import time
import hashlib
import requests
from htpasswd import HtpasswdFile

def requestApi(dataJson={}):
    API_KEY = "10000"
    API_TOKEN = "513b60e222800340e4d7a12f2454794e"

    # 提交的数据
    headers = {
        'user-agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36',
        'content-type':'application/json', 
        'x-requested-with': 'XMLHttpRequest',
    }
    # 认证提交的值参数转为字符串，否则导致验证失败
    authInfo = {
        "key": API_KEY,
        "timestamp": '%s'%int(time.time()),
        "version": '1.0'
    }
    data = ('%s|%s|%s') %(
        json.dumps(dataJson, sort_keys=True, separators=(',', ':')), 
        json.dumps(authInfo, sort_keys=True, separators=(',', ':')), 
        API_TOKEN
    )
    authInfo['auth-fields'] = ','.join(authInfo.keys())
    authInfo['sign'] = hashlib.sha256(data.encode('utf-8')).hexdigest();
    authInfo['debug'] = "1"
    for idx in authInfo:
        headers["x-api-%s"%(idx.lower())] = authInfo[idx]

    resp = requests.post("http://127.0.0.1:9005/getAccountList", data=json.dumps(dataJson), headers=headers)
    print(resp.text)


def passwdUser():
    filename = '../pwd.conf'
    username = 'xiaoguo'
    password = '123456'

    passwdfile = HtpasswdFile(filename)
    passwdfile.load()
 
    # 删除用户
    #passwdfile.delete(username)

    # 更新用户及密码
    passwdfile.update(username, password)

    # 执行完保存
    passwdfile.save()

if __name__ == '__main__':
    # passwdUser()
    requestApi()

File: example/t.php
<?php
define('API_KEY', '10000');
define('API_TOKEN', '513b60e222800340e4d7a12f2454794e');

/**
 * 数组键值排序
 */
function tksort(&$array) {
  	ksort($array);
  	foreach(array_keys($array) as $k) {
    	if(gettype($array[$k])=="array") {
      		tksort($array[$k]);
      	}
    }
}

/**
 * 调用svn client微服务接口
 */
function svnClientCall($api, $params=array())
{
	// 加密前将参数值转为字符串
	$authInfo = array(
	    'key'=>strval(API_KEY),
	    'timestamp'=>strval(time()),
	    'version'=>'1.0',
	);
	ksort($authInfo); tksort($params);
	$authInfo['sign'] = hash('sha256', strval(stripslashes(json_encode($params)).'|'.json_encode($authInfo).'|'.API_TOKEN));
	$authInfo['auth-fields'] = implode(',', array_keys($authInfo));
	$authInfo['debug'] = true;

	$content = json_encode($params);
	$headers = array(
	    "Content-Type: application/json",
	    "X-Requested-With: XMLHttpRequest",
	    "User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36",
	);
	foreach($authInfo as $key=>$val) {
		$headers[] = "X-Api-{$key}: {$val}";
	}
	// 也可以通过file_put_contents提交请求
	// $rawHeader = implode("\r\n", $headers)."\r\n";
	// $ctx = stream_context_create(array(
	//     'http' => array(
	//         'method'  => 'POST',
	//         'header'  => $rawHeader,
	//         'content' => $content
	//     )
	// ));
	// $jsonResponse = file_get_contents("http://127.0.0.1:9005/{$api}", false, $ctx);
	// $jsonResponse = file_get_contents("http://192.168.2.18:9005/{$api}", false, $ctx);

	$url = "http://127.0.0.1:9005/{$api}";
	$ch = curl_init();
	curl_setopt($ch, CURLOPT_URL, $url);
	curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, $content);
	curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
	curl_setopt($ch, CURLOPT_TIMEOUT, 60);
	$jsonResponse = curl_exec($ch);
	return @json_decode($jsonResponse, true);
}

$data = svnClientCall('getAccountList', array('a'=>'123'));
print_r($data);die;